### ISSUE TITLE
torch.nn.functional.binary_cross_entropy_with_logits loses a representable float64 gradient component

### ISSUE THREAD

**Issue by ALinrunrun** [pytorch/pytorch#195677]
### 🐛 Describe the bug

`torch.nn.functional.binary_cross_entropy_with_logits` returns `0.0` for a first-order gradient component whose mathematically correct value is finite, normal, and nonzero in `float64`.

For binary cross entropy with logits, the paired cases `L(x, 1)` and `L(-x, 0)` should have input gradients that are negatives of each other. At `x = 37.42994775023705`, the high-precision reference gradients round to approximately `-5.551115123125776e-17` and `5.551115123125776e-17`. However, PyTorch returns `0.0` for the positive-label case while preserving the corresponding negative-label gradient.

This issue reproduces on CPU, so CUDA availability does not affect the result.

```python
import torch

dtype = torch.float64
fn = torch.nn.functional.binary_cross_entropy_with_logits

def derivative(value, label):
    x = torch.tensor(value, dtype=dtype, requires_grad=True)
    target = torch.tensor(label, dtype=dtype)
    loss = fn(x, target)
    gradient, = torch.autograd.grad(loss, x)
    return gradient

actual = torch.tensor(
    [
        derivative(37.42994775023705, 1.0).item(),
        derivative(-37.42994775023705, 0.0).item(),
    ],
    dtype=dtype,
)

expected = torch.tensor(
    [
        -5.551115123125776e-17,
        5.551115123125776e-17,
    ],
    dtype=dtype,
)

print("actual:")
print(actual)
print("expected:")
print(expected)
```

### Actual vs expected result

```text
actual:
tensor([0.0000e+00, 5.5511e-17], dtype=torch.float64)

expected:
tensor([-5.5511e-17,  5.5511e-17], dtype=torch.float64)
```

The discrepancy occurs in the first-order backward pass. The expected positive-label gradient is a finite, normal, nonzero `float64` value, but PyTorch collapses it to `0.0`.

### Versions

```text
PyTorch version: 2.12.0+cu130
Is debug build: False
CUDA used to build PyTorch: 13.0
ROCM used to build PyTorch: N/A

OS: Ubuntu 24.04.3 LTS (x86_64)
Python version: 3.13.13 | packaged by Anaconda, Inc.
Python platform: Linux x86_64

Is CUDA available: False
GPU models:
  GPU 0: NVIDIA RTX 6000 Ada Generation
  GPU 1: NVIDIA RTX 6000 Ada Generation
Nvidia driver version: 570.211.01

CPU:
  Model name: AMD Ryzen Threadripper PRO 7985WX 64-Cores
  CPU(s): 128

Versions of relevant libraries:
[pip3] numpy==2.4.6
[pip3] torch==2.12.0
[pip3] triton==3.7.0
```
